pydis_site/apps/content/resources/guides/pydis-guides/contributing/bot.md

```diff
@@ -2,207 +2,643 @@
+        </span>
+    </button>
+    <div class="collapsible-content">
+        <div class="card-content">
+            The requirements for Docker are:
+            <ul>
+                <li><a href="https://docs.docker.com/install">Docker CE</a></li>
+                <li>Docker Compose. If you're using macOS and Windows, this already comes bundled with the previous installation. Otherwise, you can download it either from the <a href="https://docs.docker.com/compose/install">website</a>, or by running <code>pip install docker-compose</code>.</li>
+            </ul>
+            <p>If you get any Docker related errors, reference the <a href="../docker#possible-issues">Possible Issue</a> section of the Docker page.</p>
+        </div>
+    </div>
+</div>
+<br>
+
+Assuming you have Docker installed, enter the cloned repo in the command line and type `docker-compose up`.
+
+After pulling the images and building the containers, your bot will start. Enter your server and type `!help` (or whatever prefix you chose instead of `!`).
+
+Your bot is now running, but this method makes debugging with an IDE a fairly involved process. For additional running methods, continue reading the following sections.
+
+#### With the Bot Running Locally
+The advantage of this method is that you can run the bot's code in your preferred editor, with debugger and all, while keeping all the setup of the bot's various dependencies inside Docker.
+
+* [Prepare the hosts file on your machine](../hosts-file).
+* Append the following line to your `.env` file: `BOT_API_KEY=badbot13m0n8f570f942013fc818f234916ca531`.
+* In your `config.yml` file, set `urls.site` to `"pythondiscord.local:8000"`. If you wish to keep using `web:8000`, then [COMPOSE_PROJECT_NAME](../docker/#compose-project-names) has to be set to use this domain.
+
+You will need to start the services separately, but if you got the previous section with Docker working, that's pretty simple:
+
+* `docker-compose up web` to start the site container. This is required.
+* `docker-compose up snekbox` to start the snekbox container. You only need this if you're planning on working on the snekbox cog.
+* `docker-compose up redis` to start the Redis container. You only need this if you're not using fakeredis. For more info refer to [Working with Redis](#optional-working-with-redis).
+
+You can start several services together: `docker-compose up web snekbox redis`.
+
+##### Setting Up a Development Environment
+The bot's code is Python code like any other. To run it locally, you will need the right version of Python with the necessary packages installed:
+
+1. Make sure you have [Python 3.9](https://www.python.org/downloads/) installed. It helps if it is your system's default Python version.
+2. Install Poetry by following [these instructions](https://github.com/python-poetry/poetry#installation), or with `pip install poetry`. In case of the latter, make sure pip installs to the right Python version.
```

Comment:
I recommend not including the `pip install poetry` since it's something poetry itself discourages.